modules/misc/submodule-support.nix

```diff
@@ -15,5 +15,18 @@ with lib;
         in, for example, NixOS or nix-darwin.
       '';
     };
+
+    externalPackageInstall = mkOption {
```

Comment:
Is this option really needed though? I don't see any other use outside of NixOS, and NixOS already has `useUserPackages` for that.

Reply:
Hmm, in principle I think it could be avoided but `useUserPackages` is a NixOS option and `externalPackageInstall` is a Home Manager option. So if this option is removed the we'd need some other way to set `home.profileDirectory`, it could be set directly from the NixOS module but then `home.profileDirectory` would have to switch from `readOnly = true` to `readOnly = false`.

Keeping `externalPackageInstall` seems the most straight forward solution as far as I can tell but I'd be happy to hear of alternative solutions.